sdk/src/nova/seq.rs

```diff
@@ -153,8 +153,8 @@ impl Prover for Nova<Local> {
 impl Parameters for PP {
     type Error = Error;
 
-    fn generate_for_testing() -> Result<Self, Self::Error> {
-        Ok(gen_vm_pp(K, &()).map_err(ProofError::from)?)
+    fn generate_for_testing(k: usize) -> Result<Self, Self::Error> {
```

Comment:
You should also modify the `generate` function calls as well.

Reply:
I'm thinking instead of let user modify the value `k`, which they have no clue, I think we can check the total memory of the system and determine the best value `k` for them. What do you think ? @slumber  @sjudson

Reply:
done in 0ae46af

Reply:
I think it's a bad idea and additional maintenance. CLI defaults to `k=16` which is fine for most users.

Making it configurable like [here](https://docs.nexus.xyz/zkvm/configuring-prover#configuring-k) is sufficient.

Reply:
I agree with @slumber, I think this is overly complicated and puts far too strong assumptions on the availability of the memory. I think we can set `k=16` for now and be done with it.

Reply:
Thanks @slumber  and @sjudson , I update the default `K` from 64 to 16. in 6a1b337